03-trade-protocol.md

```diff
@@ -0,0 +1,79 @@
+* The **Taker**: issues an order which can immediately be matched with an existing (maker) order in the order book.
+
+In order to sustain the robustness and censorship resistance of the network, OpenDEX lacks a global order book and a centralized matching engine. Instead, each node on the network has its own order book and matching engine. Trading can be limited to a preferred set of trading pairs. In case of a match, the taker order node is expected to instantiate the swap protocol.
+
+## Order Type
+
+	`string id = 1`
+	The order's globally unique identifier, generated by the sender 
+
+    `string pair_id = 1`
+    A trading pair symbol, constructed with the base currency first, followed by a '/' separator and the quote currency (e.g., “LTC/BTC”)
+
+    `double price = 3`
+    The price for the order expressed in units of the quote currency
+
+    `uint64 quantity = 4`
+    The number of satoshis (or equivalent) for the order
+
+    `bool is_buy = 5`
+    Whether the order is a buy (true) or a sell (false)
+
+## Trade Protocol
+### Order Message (0x06)
+
+    `string id = 1`
+	The message's globally unique identifier, generated by the sender 
+
+	`Order order = 2`
+	The order
+
+The `Order` message is used to tell a peer about a new maker order. It should only be used after the order could not be matched locally. 
+
+### OrderInvalidation Message (0x07)
+
+	`string id = 1`
+	The message's globally unique identifier, generated by the sender 
+
+    `string order_id = 2`
+    The order’s unique identifier
+
+    `string pair_id = 3`
+    The trading pair symbol associated with the order
+
+    `uint64 quantity = 4`
+    The number of satoshis (or equivalent) to invalidate from the order sum
+
+The `OrderInvalidation` message is used to tell a peer about the invalidation of an existing order or partial order. It allows the peer to remove the order from his local order book. Order invalidation is a common event which occurs due to order cancellation or filling (by another order). Failing to send updates about the order will result in peers having a stale order in their local order books, which they may fill and then instantiate a swap procedure which is doomed to fail. In this case, the maker node’s reputation may be penalized. 
+
+### The GetOrders Message (0x08)
+
+	`string id = 1`
+	The message's globally unique identifier, generated by the sender 
+
+    `repeated string pair_ids = 2`
+    The requested orders trading pair symbols, constructed with the base currency first, followed by a  '/' separator and the quote currency (e.g., [“LTC/BTC”, “DAI/BTC”])
+
+The `GetOrders` message is used to query a peer for a list of all open orders for the specified trading pairs. It is also used to initialize the local order book with a snapshot of the peer's existing open orders. New orders are expected to get pushed by the peer via the `Order` message, instead of being queried for. 
```

Comment:
It is mainly used to initialize the local order book with a snapshot of the peer's existing open orders after establishing a connection to the peer.